python/lsst/cp/pipe/ptc/cpExtractPtcTask.py

```diff
@@ -650,8 +627,67 @@ def measureMeanVarCov(self, exposure1, exposure2, region=None):
 
         return mu, varDiff, covDiffAstier
 
-    def getGainFromFlatPair(self, exposure1, exposure2, correctionType='NONE',
-                            readNoise=None, region=None):
+    def getImageAreasMasksStats(self, exposure1, exposure2, region=None):
+        """Get image areas in a region as well as masks and statistic objects.
+
+        Parameters
+        ----------
+        exposure1 : `lsst.afw.image.exposure.ExposureF`
+            First exposure of flat field pair.
+        exposure2 : `lsst.afw.image.exposure.ExposureF`
+            Second exposure of flat field pair.
+        region : `lsst.geom.Box2I`, optional
+            Region of each exposure where to perform the calculations
+            (e.g, an amplifier).
+
+        Returns
+        -------
+        resultsDict : `dict`
+            Dictionary with masked image regions, mask planes, statistic
+            control objects, and clipped means.
+        """
+        resultsDict = {}
+        if region is not None:
+            im1Area = exposure1.maskedImage[region]
+            im2Area = exposure2.maskedImage[region]
+        else:
+            im1Area = exposure1.maskedImage
+            im2Area = exposure2.maskedImage
+
+        if self.config.binSize > 1:
+            im1Area = afwMath.binImage(im1Area, self.config.binSize)
+            im2Area = afwMath.binImage(im2Area, self.config.binSize)
+
+        im1MaskVal = exposure1.getMask().getPlaneBitMask(self.config.maskNameList)
+        im1StatsCtrl = afwMath.StatisticsControl(self.config.nSigmaClipPtc,
```

Comment:
Do you need two stat control objects that are essentially the same?  Is the concern that the mask planes will be in a different order?  These are also returned, but I'm not sure I see them used elsewhere.  Are they used in code that hasn't changed?

Reply:
Is there a guarantee that `exposure1.getMask().getPlaneBitMask(self.config.maskNameList)` will be equal to `exposure2.getMask().getPlaneBitMask(self.config.maskNameList)` ? Would that object be the same to be used in ` ratioImMaskVal = ratioIm.getMask().getPlaneBitMask(self.config.maskNameList)` of ` getGainFromFlatPair` and in ` diffImMaskVal = diffIm.getMask().getPlaneBitMask(self.config.maskNameList)` of `measureMeanVarCov`?

Reply:
As for not being used after being returned, I think you are correct, thanks!